application/src/main/java/org/togetherjava/tjbot/commands/SlashCommand.java

```diff
@@ -122,34 +124,23 @@
      * Details are available in the given event and the event also enables implementations to
      * respond to it.
      * <p>
-     * The helper {@link org.togetherjava.tjbot.commands.system.ComponentIds} can be used to
-     * generate valid IDs. Alternatively, if {@link SlashCommandAdapter} has been extended, it also
-     * offers a handy {@link SlashCommandAdapter#generateComponentId(String...)} method to ease the
-     * flow.
-     * <p>
-     * Buttons or menus have to be created with component ID (see
+     * Buttons or menus have to be created with a component ID (see
      * {@link ComponentInteraction#getComponentId()},
      * {@link net.dv8tion.jda.api.interactions.components.Button#of(ButtonStyle, String, Emoji)}) in
      * a very specific format, otherwise the command system will fail to identify the command that
      * corresponded to the button or menu click event and is unable to route it back.
      * <p>
-     * The component ID has to be a string with less than 100 characters, in a valid CSV format
-     * (separated by comma {@code ,}), where:
-     * <ul>
-     * <li>the first element is a unique integer (within the context of this message),</li>
-     * <li>the second element is a string representing the name of this command (it must be equals
-     * to the name returned by {@link #getName()}),</li>
-     * <li>and any further element can be custom strings in order to transport arbitrary data (e.g.
-     * the user id who created the button dialog)</li>
-     * </ul>
-     * An example would be {@code 13,ping,"hello,world"}.
-     * <p>
-     * Any provided extra arguments can then be picked up during the corresponding event (see
+     * The component ID has to be a UUID-string (see {@link java.util.UUID}), which is associated to
+     * a specific database entry, containing meta information about the command being executed. Such
+     * a database entry can be created and a UUID be obtained by using
+     * {@link ComponentIdGenerator#generate(ComponentId, Lifespan)}, as provided by the instance
+     * given to {@link #acceptComponentIdGenerator(ComponentIdGenerator)} during system setup. The
+     * required {@link ComponentId} instance accepts optional extra arguments, which, if provided,
+     * can be picked up during the corresponding event (see
```

Comment:
Since this is a central part of the documentation, and everything else points to it, It's maybe worth expending on it a bit.

The whole system is pretty abstract and complex, with lots of advanced java concepts. So I'm thinking, how can we make it more digestible to a newcomer?

I think a short usage example of the system would help a lot. Also, there is a ton of good information written in this PR, but it's not in the code. So I guess the best way to go about it would be to add a wiki page (slightly modified PR description) that is referenced in the code.

Reply:
Good point. I added a link to a wiki page which I will write now.